llap-server/src/java/org/apache/hadoop/hive/llap/io/api/impl/LlapIoImpl.java

```diff
@@ -151,8 +151,10 @@ private LlapIoImpl(Configuration conf) throws IOException {
       LowLevelCachePolicy
           realCachePolicy =
           useLrfu ? new LowLevelLrfuCachePolicy(minAllocSize, totalMemorySize, conf) : new LowLevelFifoCachePolicy();
-      // TODO: if realCachePolicy is not something that supports proactive caching
-      // turn the feature off (LLAP_IO_PROACTIVE_EVICTION_ENABLED) and log it
+      if (!(realCachePolicy instanceof ProactiveEvictingCachePolicy.Impl)) {
```

Comment:
It would be very cumbersome to check this on HS2 side due to potentially different config files being used for HS2 and LLAP side. Also these services might get restarted independently from each other with different config.